velox/functions/prestosql/DecimalVectorFunctions.cpp

```diff
@@ -451,6 +451,36 @@ class Round {
   }
 };
 
+class Floor {
+ public:
+  template <typename R, typename A>
+  inline static void apply(
+      R& r,
+      const A& a,
+      uint8_t /*aPrecision*/,
+      uint8_t aRescale,
+      const ExtraParams& /*param*/) {
+    auto rescaleFactor = DecimalUtil::kPowersOfTen[aRescale];
+    auto increment = (a % rescaleFactor) < 0 ? -1 : 0;
```

Comment:
`auto increment = (a < 0) ? -1 : 0;` should do?

Reply:
Thanks for the suggestion, modifying it to `(a < 0)` results in different results from Presto for cases where `a % rescaleFactor` is `0`. The increment calculation is borrowed from Presto:
```
long rescaleFactor = Decimals.longTenToNth((int) numScale);
long increment = (num % rescaleFactor) < 0 ? -1 : 0;
```

The following query for instance returns `-124` in velox with `(a < 0)` but returns `-123` in Presto:
```
presto> select floor(CAST(DECIMAL '-123.00' AS DECIMAL(5,2)));
 _col0
-------
 -123
```
Could you please share how to proceed?

Reply:
Thanks!